planning/behavior_velocity_run_out_module/src/dynamic_obstacle.cpp

```diff
@@ -312,6 +312,31 @@ void calculateMinAndMaxVelFromCovariance(
   dynamic_obstacle.max_velocity_mps = max_velocity;
 }
 
+double convertDurationToDouble(const builtin_interfaces::msg::Duration & duration)
+{
+  return duration.sec + duration.nanosec / 1e9;
+}
+
+// Create a path leading up to a specified prediction time
+std::vector<geometry_msgs::msg::Pose> createPathToPredictionTime(
+  const autoware_auto_perception_msgs::msg::PredictedPath & predicted_path, double prediction_time)
+{
+  // Calculate the number of poses to include based on the prediction time and the time step between
+  // poses
+  const double time_step_seconds = convertDurationToDouble(predicted_path.time_step);
+  const size_t poses_to_include =
+    std::min(static_cast<size_t>(prediction_time / time_step_seconds), predicted_path.path.size());
```

Comment:
I think we can assume time_step_seconds is positive non-zero value

Reply:
I added the handling when time_step is too close to zero just in case!
https://github.com/autowarefoundation/autoware.universe/pull/6650/commits/7e05b9891cc5efa840d47d73d63fb5401ed921a6